src/D2L.CodeStyle.TestAnalyzers/NUnit/CategoryAnalyzer.cs

```diff
@@ -221,24 +239,29 @@ ISymbol symbol
 
 		private static bool TryLoadNUnitTypes(
 			Compilation compilation,
-			out NUnitTypes types
+			[NotNullWhen( true )]
+			out NUnitTypes? types
 		) {
-			INamedTypeSymbol categoryAttribute = compilation.GetTypeByMetadataName( "NUnit.Framework.CategoryAttribute" );
+			INamedTypeSymbol? categoryAttribute = compilation.GetTypeByMetadataName( "NUnit.Framework.CategoryAttribute" );
 			if( categoryAttribute == null || categoryAttribute.TypeKind == TypeKind.Error ) {
 				types = null;
 				return false;
 			}
 
-			ImmutableHashSet<INamedTypeSymbol> testAttributes = ImmutableHashSet
-				.Create<INamedTypeSymbol>(
-					SymbolEqualityComparer.Default,
-					compilation.GetTypeByMetadataName( "NUnit.Framework.TestAttribute" ),
-					compilation.GetTypeByMetadataName( "NUnit.Framework.TestCaseAttribute" ),
-					compilation.GetTypeByMetadataName( "NUnit.Framework.TestCaseSourceAttribute" ),
-					compilation.GetTypeByMetadataName( "NUnit.Framework.TheoryAttribute" )
-				);
+			ImmutableHashSet<INamedTypeSymbol> testAttributes = new[] {
+				compilation.GetTypeByMetadataName( "NUnit.Framework.TestAttribute" ),
+				compilation.GetTypeByMetadataName( "NUnit.Framework.TestCaseAttribute" ),
+				compilation.GetTypeByMetadataName( "NUnit.Framework.TestCaseSourceAttribute" ),
+				compilation.GetTypeByMetadataName( "NUnit.Framework.TheoryAttribute" )
+			}.Where( x => x != null )!

```

Comment:
Is the null-forgiving op necessary on the result of `Where`?

Reply:
Yep, well known problem.

They've talked about extending nullable analysis more to handle LINQ stuff but nothing very concrete. Also related to covariance headaches (e.g. `Task<T>` vs `Task<T?>`)

The `Where( x => x is not null )!` thing is pretty common from what I've seen. Sometimes people make their own extensions. I've also seen `.OfType<T>()` but that feels even worse to me (not obvious what kind of "casting" is happening there)